src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,48 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.lang.reflect.Array;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_SIZE = 10;
+
+    private final K[] keys;
+    private final V[] values;
+    private int space;
+
+    public StorageImpl() {
+        keys = (K[]) Array.newInstance(Object.class, STORAGE_SIZE);
+        values = (V[]) Array.newInstance(Object.class, STORAGE_SIZE);
+    }
+
+    public boolean verify(K k, K v) {
+        return (k != null && k.equals(v) || k == v);
+    }
```

Comment:
```suggestion
    private boolean areKeysEqual(K key1, K key2) {
        return key1 == key2 || key1 != null && key1.equals(key2);
    }
```
Let's declare this method after public ones.